spec/filetypes/ruby_spec.lua

```diff
@@ -218,4 +218,43 @@ describe("array", function()
       })
     )
   end)
+
+  it("collapses multi-line array and skips embedded comments", function()
+    assert.are.same(
+      { "[1, 2, 3]" },
+      Helper:call({
+        "[ # a",
+        "  1, # b",
+        "  2, # c",
+        "=begin",
+        "a multiline comment here",
+        "and one more line",
+        "=end",
+        "  3 # d",
+        "]"
+      })
+    )
+  end)
```

Comment:
I'm not sure about throwing out the comment altogether. I think it would be more reasonable to just bail and not change the buffer, like when attempting to collapse a multi-line block into a single line.